docs/design/proposals/codegen-refactor/model_cmd_2.md

```diff
@@ -0,0 +1,143 @@
+* The cache is only used to store the `aws-sdk` repo
+* The red line denotes a dependency on $SERVICE-controller (output directory) repo that should be removed
+* Only `GetCRDs` is duplicated between the commands when creating the model
+
+
+Current Dev Loop
+---
+
+![current-dev-loop](./images/current_dev_loop.png)
+* Validating the result of a change in *code-generator* and/or *generator.yaml* can be a lengthy process
+* In the current feedback loop, users and contributors need to manually verify files or run lengthy e2e tests even for minor changes to `ackgenconfig`
+
+
+## Solution
+Add a new command, `model`, which takes in `aws-sdk` and *generator.yaml* as inputs to create and cache `ackmodel`. Existing commands, `api` and `controller`, will remove parameters for `aws-sdk` and *generator.yaml* and replace them with `ackmodel` input and/or let it default to `--cache-dir`.
+
+Updated Pipeline
+---
+![updated-pipeline](./images/proposed_pipeline.png)
+* This is not an exhaustive diagram of the calls, but shows the clear responsibility of `generateModel` and how downstream commands like `api` and `controller` become significantly lighter and easier to follow.
+* `ackmodel` will be cached in the same folder as `aws-sdk` and downstream commands will check for `ackmodel` instead of `aws-sdk`
+* users can override with a path to their own `ackmodel` (for even quicker testing!), but by default will check `--cach-dir`
+* No more dependency on $SERVICE-controller repo. The `ackmodel` will be generated with a specific version which can be extracted after commands `GetACKModel()`
+
+Updated Dev Loop:
+---
+![updated-dev-loop](./images/proposed_dev_loop.png)
+* Diffing the `ackmodel` between *code-generator* runs will be quicker than manually checking files or running e2e tests
+* Users can directly edit `ackmodel` or underlying `aws-sdk` / `ackgenconfig` then generate $SERVICE-controller for even quicker iteration loops
+
+
+Updated `ackmodel` creation:
+---
+To address the inconsistent `ackmodel` creation, I propose we edit `aws-sdk` directly as much as we can, then introduce more fields to represent other relations that can't be applied by editing `aws-sdk`. Therefore, if we want to rename a field/resource, it would be instead completely removed from model. Conceptually:
+
+```
+func (m *Model) RenameShapes() error {
+  for shapeName, shape := range m.SDKAPI.API.Shapes {
+    if m.cfg.shapeRenamed(shapeName) {
+      updateName := m.cfg.GetShapeName(shapeName)
+      m.SDKAPI.Shapes[updatedName] = shape
+      delete(shapeName, shape)
+    }
+  }
+}
+```
+
+Downstream funcs like `SetResource` become easier to use because iterating like so will already have `ackgenconfig` applied:
+
+```
+updatedShape := m.GetOutputShape(outputShape)
+
+for memberIndex, memberName := range updatedShape.MemberNames() {
+  // updatedShape already has its fields renamed, removed, or added to based
+  ...
+}
+
+```
```

Comment:
++ I love this

Reply:
`aws-sdk-go` is edited and part of `ackmodel` in this proposal/example not an immutable version. More info in comment above